### PR TITLE
JAMES-2366 Introduce Forward RRT mapping type 

### PR DESCRIPTION
**Pull request by aduprat**
Rebased & squashed version of #1348

### CODE REVIEW

File: server/data/data-api/src/main/java/org/apache/james/rrt/api/RecipientRewriteTableManagementMBean.java [hunk 1]
```diff
@@ -187,4 +187,30 @@
      *            If an error occurred
      */
     Map<String, Mappings> getAllMappings() throws Exception;
+
+    /***
+     * Add forward mapping
+     *
+     * @param toUser
```
Comment (mbaechler) [inline server/data/data-api/src/main/java/org/apache/james/rrt/api/RecipientRewriteTableManagementMBean.java]: javadoc params don't match method params

File: server/data/data-api/src/main/java/org/apache/james/rrt/api/RecipientRewriteTableManagementMBean.java [hunk 2]
```diff
@@ -187,4 +187,30 @@
...
+     *            the username part of the mail address destination defined for this forward.
+     * @param toDomain
+     *            the domain part of the mail address destination defined for this forward.
+     * @param fromAddress The base address of the forward. Mails for this address will be sent to the added forward destination.
+     *            the address.
+     * @throws Exception If an error occurred
+     */
+    void addForwardMapping(String user, String domain, String address) throws Exception;
```
Comment (mbaechler) [inline server/data/data-api/src/main/java/org/apache/james/rrt/api/RecipientRewriteTableManagementMBean.java]: why don't we use an address for both part ?

File: server/data/data-api/src/main/java/org/apache/james/rrt/lib/Mapping.java [hunk 3]
```diff
@@ -47,16 +47,18 @@ static Type detectType(String input) {
...
+        Domain("domain:", 1),
+        Error("error:", 3),
+        Forward("forward:", 2),
+        Address("", 3);
 
         private final String asPrefix;
+        private final int order;
 
-        Type(String asPrefix) {
+        Type(String asPrefix, Integer order) {
```
Comment (mbaechler) [inline server/data/data-api/src/main/java/org/apache/james/rrt/lib/Mapping.java]: why not `int` ?

**Comment by chibenwa** [comment]
See https://github.com/linagora/james-project/pull/1364

**Pull request closed without merging**